source/concepts/applications.rst

```diff
@@ -0,0 +1,746 @@
+by setting the ``imd.velocity_reset_available`` key to true in the :ref:`shared
+state <state-service>`.
+
+Sending user interactions
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Users send, on the :ref:`shared state <state-service>`, the description of the
+interactions they want to apply. There is no limit to the number of interaction
+a user can send. Each interaction is described under the key
+``interaction.<INTERACTION_ID>`` where ``<INTERACTION_ID>`` is an arbitrary
+string, unique to the interaction, used to identify it. It is commonly a UUID4.
+Under that key, the value is a Struct with the following keys:
+
+* ``positions``: the coordinates of the interaction's origin in simulation
+  space. This is typically a position attached to the controller of the user in
+  VR, but it does not have to be. By default, this is `[0, 0, 0]`.
+* ``particles``: the indices of the affected particles in the array of
+  particles used by the :ref:`trajectory application <trajectory-application>`.
+  If the order in this array does not match the order used by the simulation
+  engine, it is the server's responsibility to map them. The default value is
+  an empty list.
+* ``type``: the kind of interaction to apply, this is what defines what
+  :ref:`force equation <force-equations>` will be used. It should be set to
+  `gaussian` for the Gaussian force, `spring` for the harmonic force, and
+  `constant` for the constant force. Interactions with an type unknown to the
+  server will be ignored silently. By default, the Gaussian force is assumed.
+* ``scale``: the scaling factor :math:`s` to apply to the force. The default
+  scale is 1.
+* ``mass_weighted``: a boolean, true if the interaction is mass weighted, false
+  otherwise. The default is true.
+* ``max_force``: the maximum force magnitude that can be applied to a particle
+  by this interaction. The default is 20,000
+  :math:`kJ\cdot\text{mol}^{-1}\cdot\text{nm}^{-1}`.
+* ``reset_velocities``: a boolean, true if :ref:`velocity reset
+  <velocity-reset>` should be applied, false otherwise. This is false by
+  default and will be ignored silently is the server does not have the feature.
+
+If the iMD application is used in conjunction with the :ref:`multiplayer
+application <multiplayer-application>`, then the interaction can also use the
+following fields:
+
+* ``owner.id``: if the interaction originates from a client that defines an
+  avatar, it can set this field to the player id attached to its avatar. This
+  allows to match interactions with avatars when analysing session recordings.
+* ``label``: used with ``owner.id``, this is the name of the avatar component
+  from which the interaction originates (`e.g.` ``hand.right`` or
+  ``hand.left``).
+
+.. _imd-framedata-keys:
+
+FrameData keys
+~~~~~~~~~~~~~~
+
+Some details about how the user interactions where applied can be added to the
+:ref:`FrameData <frame-description>`.
+
+The sum of the energies from user interactions can be included, in
```

Comment:
Is it worth specifying how the total energy is calculated, and warning them here (as well as elsewhere) that this corresponds to whatever the kinetic and potential energies output by the simulation engine which may correspond to different times in a given frame?

Reply:
I do not understand what yu mean by "which may correspond to different times in a given frame"

Reply:
Ah I'm sorry for not explaining this well. I am referring to the difficulties with calculating total energies when using a leap-frog algorithm for integration such that the kinetic energy stored in a frame is half a time step behind the potential energy (with the velocities calculated as v(t - dt/2) and the positions calculated at r(t)) for a given frame